cypress/integration/battleship_spec.js

```diff
@@ -0,0 +1,144 @@
+describe('Battleship Test', () => {
+    context('responsive', () => {
+        beforeEach(() => {
+            cy.viewport('iphone-6');
+        });
+
+        it('Opens the welcome page', () => {
+            cy.visit('/');
+            cy.contains('Welcome to the Battleship!');
+        });
+
+        it('Shows the opponent fleet', () => {
+            cy.visit('/');
+            cy.contains('Create new game').click();
+
+            cy.contains('Voir la flotte Adverse').click();
+            cy.get('.ship').should('not.exist');
+        });
+
+        it('Plays a complete game', () => {
+            let game;
+            let player1;
+            let player2;
+            let currentPlayer;
+
+            cy.request(
+                'POST',
+                `${Cypress.env('api_server')}/games?mode=demo`
+            ).then(response => {
+                game = response.body.game_hash;
+                player1 = response.body.player_hash;
+                currentPlayer = player1;
+
+                cy.request(
+                    'PUT',
+                    `${Cypress.env('api_server')}/games/${game}/join?mode=demo`
+                ).then(response => {
+                    player2 = response.body.player_hash;
+
+                    cy.request(
+                        'GET',
+                        `${Cypress.env(
+                            'api_server'
+                        )}/games/${game}?player_hash=${player2}`
+                    ).then(response => {
+                        cy.visit(`/games/${game}?player_hash=${player1}`);
+                        cy.contains('Voir la flotte Adverse').click();
+
+                        shoot(16);
```

Comment:
It's not easily readable. But I don't know how to do it better.

Reply:
Agree. Maybe later